tests/profiler/test_profiler.py

```diff
@@ -431,6 +432,36 @@ def test_pytorch_profiler_nested(tmpdir):
     assert events_name == expected, (events_name, torch.__version__, platform.system())
 
 
+def test_pytorch_profiler_loggercollection(tmpdir):
+    """
+    Tests whether the PyTorch profiler is able to write its trace locally when
+    the Trainer's logger is an instance of LoggerCollection. See issue #8157.
+    """
+
+    def look_for_trace(trace_dir):
+        """ Determines if a directory contains a PyTorch trace """
+        return any(("trace.json" in filename for filename in os.listdir(trace_dir)))
+
+    # Sanity check
+    assert not look_for_trace(tmpdir)
+
+    model = BoringModel()
+
+    # Wrap the logger in a list so it becomes a LoggerCollection
+    logger = [TensorBoardLogger(save_dir=tmpdir)]
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        profiler="pytorch",
+        logger=logger,
+        limit_train_batches=5,  # it takes a while for the warning to appear
+        max_epochs=1,
+    )
+
+    trainer.fit(model)
+
```

Comment:
Same here. Add a check for `assert isinstance(trainer.logger, LoggerCollection)`

Reply:
Done!